### PR TITLE
cleanup spin lock

### PR DESCRIPTION
**Pull request by mingqiangchi**
optimize the spin lock usage

Tracked-On: #4958

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
8 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write64 offset:86<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write64<br><2> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write32 offset:97<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write32<br><3> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write16 offset:108<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write16<br><4> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write8 offset:119<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write8<br><5> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read64 offset:131<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read64<br><6> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read32 offset:142<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read32<br><7> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read16 offset:153<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read16<br><8> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read8 offset:164<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read8<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/arch/x86/guest/ucode.c [hunk 1]
```diff
@@ -48,7 +47,6 @@ void acrn_update_ucode(struct acrn_vcpu *vcpu, uint64_t v)
 	int32_t err;
 	uint32_t err_code;
 
-	spinlock_obtain(&micro_code_lock);
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/ucode.c:51]: Add a comment to say the user (guest operating system) should grantee it won't issue 2nd  microcode update comment when the 1st microcode update is on going..

File: hypervisor/hw/pci.c [hunk 2]
```diff
@@ -162,20 +166,21 @@ static uint32_t pci_mmcfg_read_cfg(union pci_bdf bdf, uint32_t offset, uint32_t
...
 /*
  * @pre bytes == 1U || bytes == 2U || bytes == 4U
+ * @pre offset 8-byte alignment  if byte == 1U
+ *             16-byte alignment if byte == 2U
+ *             32-byte alignment if byte == 4U
  */
 static void pci_mmcfg_write_cfg(union pci_bdf bdf, uint32_t offset, uint32_t bytes, uint32_t val)
 {
 	uint32_t addr = mmcfg_off_to_address(bdf, offset);
 	void *hva = hpa2hva(addr);
```
Comment (dongyaozu) [inline hypervisor/hw/pci.c:183]: ditto

File: hypervisor/arch/x86/guest/vm.c [hunk 3]
```diff
@@ -845,7 +845,7 @@ void update_vm_vlapic_state(struct acrn_vm *vm)
 	}
 
 	vm->arch_vm.vlapic_state = vlapic_state;
-	spinlock_release(&vm->vm_lock);
+	spinlock_release(&vm->vlapic_state_lock);
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vm.c:848]: Let us rename vlapic_state to vlapic_mode first, and then this lock to vlapic_mode_lock

**Comment by HuiHuangShi** [comment]
8 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write64 offset:86<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write64<br><2> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write32 offset:97<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write32<br><3> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write16 offset:108<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write16<br><4> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write8 offset:119<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write8<br><5> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read64 offset:131<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read64<br><6> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read32 offset:142<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read32<br><7> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read16 offset:153<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read16<br><8> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read8 offset:164<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read8<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by wenlingz** [comment]
@mingqiangchi  please rebase your code, thanks

**Comment by wenlingz** [comment]
@mingqiangchi , please refer:
Applying: remove dead code in io.h
Applying: cleanup spin lock in irq.c
Applying: cleanup spin lock in vtd.c
Applying: remove spin lock for micro code update
Applying: remove pci_device_lock in pci.c
error: patch failed: hypervisor/hw/pci.c:44
error: hypervisor/hw/pci.c: patch does not apply
Patch failed at 0005 remove pci_device_lock in pci.c
Use 'git am --show-current-patch' to see the failed patch
When you have resolved this problem, run "git am --continue"

**Comment by mingqiangchi** [comment]
@dongyaozu  have updated the patches.

**Comment by HuiHuangShi** [comment]
8 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write64 offset:86<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write64<br><2> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write32 offset:97<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write32<br><3> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write16 offset:108<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write16<br><4> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write8 offset:119<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write8<br><5> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read64 offset:131<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read64<br><6> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read32 offset:142<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read32<br><7> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read16 offset:153<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read16<br><8> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read8 offset:164<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read8<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
8 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write64 offset:86<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write64<br><2> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write32 offset:97<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write32<br><3> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write16 offset:108<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write16<br><4> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write8 offset:119<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write8<br><5> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read64 offset:131<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read64<br><6> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read32 offset:142<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read32<br><7> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read16 offset:153<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read16<br><8> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read8 offset:164<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read8<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
11 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/hw/pci.c func:pci_is_valid_access_offset offset:57<br> reason:Static procedure is not explicitly called in code analysed. : pci_is_valid_access_offset<br><2> config:sharing_config filename:/hypervisor/hw/pci.c func:pci_is_valid_access_offset offset:57<br> reason:Procedure is not called or referenced in code analysed. : pci_is_valid_access_offset<br><3> config:part_config filename:/hypervisor/hw/pci.c func:pci_is_valid_access_offset offset:57<br> reason:Procedure is not called or referenced in code analysed. : pci_is_valid_access_offset<br><4> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write64 offset:86<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write64<br><5> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write32 offset:97<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write32<br><6> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write16 offset:108<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write16<br><7> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write8 offset:119<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write8<br><8> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read64 offset:131<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read64<br><9> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read32 offset:142<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read32<br><10> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read16 offset:153<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read16<br><11> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read8 offset:164<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read8<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
9 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/include/hw/pci.h func:pci_is_valid_access offset:374<br> reason:Static procedure is not explicitly called in code analysed. : pci_is_valid_access<br><2> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write64 offset:86<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write64<br><3> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write32 offset:97<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write32<br><4> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write16 offset:108<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write16<br><5> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_write8 offset:119<br> reason:Static procedure is not explicitly called in code analysed. : mmio_write8<br><6> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read64 offset:131<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read64<br><7> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read32 offset:142<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read32<br><8> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read16 offset:153<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read16<br><9> config:common filename:/hypervisor/include/arch/x86/io.h func:mmio_read8 offset:164<br> reason:Static procedure is not explicitly called in code analysed. : mmio_read8<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**